supervisor.go

```diff
@@ -0,0 +1,123 @@
+package streams
+
+// Supervisor represents a commit supervisor.
+//
+// The Supervisor performs a commit in a concurrently-safe manner.
+// There can only ever be 1 ongoing commit at any given time.
+type Supervisor interface {
+	// Perform a global commit sequence.
+	//
+	// If triggered by a Node, that node should be passed as an argument to Commit.
```

Comment:
`If triggered by a Pipe, the associated Processor should be passed.`

Reply:
👍